test/ha/activator_test.go

```diff
@@ -49,7 +50,7 @@ func TestActivatorHAGraceful(t *testing.T) {
 
 func TestActivatorHANonGraceful(t *testing.T) {
 	// For non-graceful tests, we want the pod to receive a SIGKILL straight away.
-	testActivatorHA(t, ptr.Int64(0), 0.95)
+	testActivatorHA(t, ptr.Int64(0), 1-1.0/float64(haReplicas))
```

Comment:
@mattmoor  I find that targeting slo of 50% when we run this test with 2 activators (I think we specifically patch this with 2) is equivalent to basically not running the test at all :)

Reply:
/hold cancel

Not the hill I'm going to die on.  We can keep lowering this number as we see flakes, but I suspect the only reason it's above `(1-1/replicas)` is due to the sleeps, which are likely padding that bound with 100% `200 OK`s for a few seconds 🙃 

> is equivalent to basically not running the test at all

I think the pertinent question is: what is this test actually testing?  The goal isn't an SLO of 100%, since we aren't shutting down gracefully, so realistically what do we expect the capacity reduction to be?  Why would we make the test measure for more than that?  Other than to torture ourselves! 😉 

My understanding is that it is to check that the temporary reduction in availability as K8s heals is proportional to the loss of capacity (until that capacity is removed / replaced).  I think it's still theoretically possible for us to miss the bound above due to unfortunate probabilistic load balancing 😬 

So in a sense, you are right: we are testing that K8s works!  But perhaps what we're really testing is that we are holding K8s right (so not useless?). 🤷


👋 See y'all next week 🏝️ 😎 🎆

Reply:
I agree it's mostly testing we're holding it right I think.